dask_expr/tests/test_collection.py

```diff
@@ -769,3 +769,21 @@ def test_align_unknown_partitions():
     df = from_pandas(pdf, npartitions=2, sort=False)
     with pytest.raises(ValueError, match="Not all divisions"):
         df.align(df)
+
+
+def test_assign_simplify(pdf):
+    df = from_pandas(pdf)
+    df2 = from_pandas(pdf)
+    df["new"] = df.x > 1
+    result = optimize(df[["x", "new"]], fuse=False)
+    expected = optimize(df2[["x"]].assign(new=df2.x > 1), fuse=False)
+    assert result._name == expected._name
+
+
+def test_assign_simplify_series(pdf):
+    df = from_pandas(pdf)
+    df2 = from_pandas(pdf)
+    df["new"] = df.x > 1
+    result = optimize(df.new, fuse=False)
+    expected = optimize(df2[[]].assign(new=df2.x > 1).new, fuse=False)
+    assert result._name == expected._name
```

Comment:
Do we want to use `assert_eq` anywhere, or do you feel that the result of `assign` is already well tested?

Reply:
Probably tested reasonably well, but the actual check is cheap so shouldn't hurt to add it